The logs are too convoluted. They should be more concise and informative. There's no need to show a full traceback when a server fails to connect, nor print that you're "reverting to the legacy protocol". Here's a snippet for reference:

```
[00m [21:10:13] [SC4MP/Main] [INFO] SC4MP API
[00m [21:10:14] [SC4MP/Main] [INFO] Starting scanner...
[00m [21:10:14] [SC4MP/Main] [INFO] Starting webserver...[90m [21:10:14] [SC4MP/Thread-2] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7240...

[00m [21:10:14] [SC4MP/Main] [INFO] Webserver started on http://localhost:8080
[00m [21:10:14] [SC4MP/Main] (Flask) [INFO] [00m [21:10:14] [SC4MP/Main] (Flask) [INFO] [00m [21:10:14] [SC4MP/Main] (Flask) [INFO]  * Serving Flask app 'sc4mpapi'
[00m [21:10:14] [SC4MP/Main] (Flask) [INFO]  * Debug mode: off
[90m [21:10:14] [SC4MP/Thread-2] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:14] [SC4MP/Thread-3] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7241...
[90m [21:10:14] [SC4MP/Thread-3] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:14] [SC4MP/Thread-4] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7242...
[90m [21:10:14] [SC4MP/Thread-4] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:14] [SC4MP/Thread-5] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7243...
[90m [21:10:14] [SC4MP/Thread-5] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:14] [SC4MP/Thread-6] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7244...
[90m [21:10:14] [SC4MP/Thread-7] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7245...
[90m [21:10:14] [SC4MP/Thread-8] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7246...
[90m [21:10:14] [SC4MP/Thread-9] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7247...
[90m [21:10:14] [SC4MP/Thread-9] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:14] [SC4MP/Thread-10] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7248...
[90m [21:10:14] [SC4MP/Thread-11] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7249...
[90m [21:10:15] [SC4MP/Thread-12] (Fetcher) [INFO] Fetching server at 163.252.250.125:7248...
[90m [21:10:15] [SC4MP/Thread-13] (Fetcher) [INFO] Fetching server at 138.201.94.174:7240...
[90m [21:10:15] [SC4MP/Thread-14] (Fetcher) [INFO] Fetching server at 35.151.248.28:7240...
[90m [21:10:15] [SC4MP/Thread-7] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.[90m [21:10:15] [SC4MP/Thread-6] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.

[90m [21:10:15] [SC4MP/Thread-15] (Fetcher) [INFO] Fetching server at 163.252.250.125:7251...
[90m [21:10:15] [SC4MP/Thread-13] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:15] [SC4MP/Thread-10] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.[90m [21:10:15] [SC4MP/Thread-15] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.

[90m [21:10:15] [SC4MP/Thread-8] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:15] [SC4MP/Thread-12] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:15] [SC4MP/Thread-16] (Fetcher) [INFO] Fetching server at 136.35.137.78:7240...
[90m [21:10:15] [SC4MP/Thread-17] (Fetcher) [INFO] Fetching server at 163.252.250.125:7241...
[90m [21:10:15] [SC4MP/Thread-16] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:15] [SC4MP/Thread-17] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:15] [SC4MP/Thread-18] (Fetcher) [INFO] Fetching server at 163.252.250.125:7243...
[90m [21:10:16] [SC4MP/Thread-19] (Fetcher) [INFO] Fetching server at 163.252.250.125:7246...
[90m [21:10:16] [SC4MP/Thread-19] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.[90m [21:10:16] [SC4MP/Thread-18] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.

[91m [21:10:17] [SC4MP/Thread-19] (Fetcher) [ERROR] [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_xwmcoo_5\\Regions\\Downloads\\config.bmp'

Traceback (most recent call last):
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 402, in _calculate_region_stats
    region_dimensions = get_bitmap_dimensions(region_config_path)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 112, in get_bitmap_dimensions
    with open(filename, "rb") as file:
         ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_xwmcoo_5\\Regions\\Downloads\\config.bmp'

[90m [21:10:17] [SC4MP/Thread-20] (Fetcher) [INFO] Fetching server at 24.128.151.227:7240...
[91m [21:10:17] [SC4MP/Thread-8] (Fetcher) [ERROR] [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_ostbjatq\\Regions\\Downloads\\config.bmp'

Traceback (most recent call last):
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 402, in _calculate_region_stats
    region_dimensions = get_bitmap_dimensions(region_config_path)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 112, in get_bitmap_dimensions
    with open(filename, "rb") as file:
         ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_ostbjatq\\Regions\\Downloads\\config.bmp'

[90m [21:10:17] [SC4MP/Thread-21] (Fetcher) [INFO] Fetching server at 163.252.250.125:7240...
[90m [21:10:17] [SC4MP/Thread-21] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:18] [SC4MP/Thread-22] (Fetcher) [INFO] Fetching server at 163.252.250.125:7245...
[90m [21:10:18] [SC4MP/Thread-22] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:18] [SC4MP/Thread-23] (Fetcher) [INFO] Fetching server at 163.252.250.125:7247...
[90m [21:10:18] [SC4MP/Thread-23] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:19] [SC4MP/Thread-24] (Fetcher) [INFO] Fetching server at 163.252.250.125:7242...
[90m [21:10:19] [SC4MP/Thread-24] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:20] [SC4MP/Thread-25] (Fetcher) [INFO] Fetching server at 163.252.250.125:7244...
[90m [21:10:20] [SC4MP/Thread-25] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:10:23] [SC4MP/Thread-26] (Fetcher) [INFO] Fetching server at 67.59.70.25:7240...
[90m [21:10:25] [SC4MP/Thread-26] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection refused by remote host.
[91m [21:10:28] [SC4MP/Thread-26] (Fetcher) [ERROR] [WinError 10061] No connection could be made because the target machine actively refused it

Traceback (most recent call last):
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 299, in run
    server_id = self.get("server_id")
                ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 359, in get
    s = self.socket_0_8()
        ^^^^^^^^^^^^^^^^^
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 352, in socket_0_8
    s.connect(self.server)
ConnectionRefusedError: [WinError 10061] No connection could be made because the target machine actively refused it

[90m [21:10:35] [SC4MP/Thread-11] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection timed out.
[90m [21:10:36] [SC4MP/Thread-14] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection timed out.
[90m [21:10:38] [SC4MP/Thread-20] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection timed out.
[93m [21:10:57] [SC4MP/Thread-11] (Fetcher) [WARNING] Server timed out.
[93m [21:10:57] [SC4MP/Thread-14] (Fetcher) [WARNING] Server timed out.
[93m [21:10:59] [SC4MP/Thread-20] (Fetcher) [WARNING] Server timed out.
[90m [21:12:02] [SC4MP/Thread-27] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7240...
[90m [21:12:02] [SC4MP/Thread-27] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:02] [SC4MP/Thread-28] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7241...
[90m [21:12:02] [SC4MP/Thread-28] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:02] [SC4MP/Thread-29] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7242...
[90m [21:12:02] [SC4MP/Thread-29] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:02] [SC4MP/Thread-30] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7243...
[90m [21:12:02] [SC4MP/Thread-31] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7244...
[90m [21:12:02] [SC4MP/Thread-32] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7245...
[90m [21:12:02] [SC4MP/Thread-33] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7246...
[90m [21:12:02] [SC4MP/Thread-34] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7247...
[90m [21:12:03] [SC4MP/Thread-35] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7248...
[90m [21:12:03] [SC4MP/Thread-36] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7249...
[90m [21:12:03] [SC4MP/Thread-37] (Fetcher) [INFO] Fetching server at 163.252.250.125:7248...
[90m [21:12:03] [SC4MP/Thread-37] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:03] [SC4MP/Thread-38] (Fetcher) [INFO] Fetching server at 138.201.94.174:7240...
[90m [21:12:03] [SC4MP/Thread-39] (Fetcher) [INFO] Fetching server at 35.151.248.28:7240...
[90m [21:12:03] [SC4MP/Thread-40] (Fetcher) [INFO] Fetching server at 163.252.250.125:7251...
[90m [21:12:03] [SC4MP/Thread-41] (Fetcher) [INFO] Fetching server at 136.35.137.78:7240...
[90m [21:12:03] [SC4MP/Thread-42] (Fetcher) [INFO] Fetching server at 163.252.250.125:7241...
[90m [21:12:04] [SC4MP/Thread-43] (Fetcher) [INFO] Fetching server at 163.252.250.125:7243...
[90m [21:12:04] [SC4MP/Thread-43] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:04] [SC4MP/Thread-44] (Fetcher) [INFO] Fetching server at 163.252.250.125:7246...
[90m [21:12:04] [SC4MP/Thread-44] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:04] [SC4MP/Thread-38] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:04] [SC4MP/Thread-42] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.[90m [21:12:04] [SC4MP/Thread-40] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.

[90m [21:12:04] [SC4MP/Thread-41] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[91m [21:12:04] [SC4MP/Thread-44] (Fetcher) [ERROR] [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_re8d7_n8\\Regions\\Downloads\\config.bmp'

Traceback (most recent call last):
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 402, in _calculate_region_stats
    region_dimensions = get_bitmap_dimensions(region_config_path)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 112, in get_bitmap_dimensions
    with open(filename, "rb") as file:
         ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_re8d7_n8\\Regions\\Downloads\\config.bmp'

[90m [21:12:05] [SC4MP/Thread-45] (Fetcher) [INFO] Fetching server at 24.128.151.227:7240...
[90m [21:12:05] [SC4MP/Thread-46] (Fetcher) [INFO] Fetching server at 163.252.250.125:7240...
[90m [21:12:05] [SC4MP/Thread-46] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:06] [SC4MP/Thread-47] (Fetcher) [INFO] Fetching server at 163.252.250.125:7242...
[90m [21:12:07] [SC4MP/Thread-48] (Fetcher) [INFO] Fetching server at 163.252.250.125:7247...
[90m [21:12:07] [SC4MP/Thread-47] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:07] [SC4MP/Thread-48] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:07] [SC4MP/Thread-49] (Fetcher) [INFO] Fetching server at 163.252.250.125:7245...
[90m [21:12:07] [SC4MP/Thread-49] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:09] [SC4MP/Thread-50] (Fetcher) [INFO] Fetching server at 163.252.250.125:7244...
[90m [21:12:09] [SC4MP/Thread-50] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:09] [SC4MP/Thread-30] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:12:09] [SC4MP/Thread-33] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.[90m [21:12:09] [SC4MP/Thread-32] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.[90m [21:12:09] [SC4MP/Thread-31] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.[90m [21:12:09] [SC4MP/Thread-35] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.



[90m [21:12:09] [SC4MP/Thread-34] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[91m [21:12:11] [SC4MP/Thread-33] (Fetcher) [ERROR] [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_6a6jnyi9\\Regions\\Downloads\\config.bmp'

Traceback (most recent call last):
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 402, in _calculate_region_stats
    region_dimensions = get_bitmap_dimensions(region_config_path)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 112, in get_bitmap_dimensions
    with open(filename, "rb") as file:
         ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_6a6jnyi9\\Regions\\Downloads\\config.bmp'

[90m [21:12:11] [SC4MP/Thread-51] (Fetcher) [INFO] Fetching server at 67.59.70.25:7240...
[90m [21:12:14] [SC4MP/Thread-51] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection refused by remote host.
[93m [21:12:22] [SC4MP/Thread-51] (Fetcher) [WARNING] Server timed out.
[90m [21:12:24] [SC4MP/Thread-36] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection timed out.
[90m [21:12:24] [SC4MP/Thread-39] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection timed out.
[90m [21:12:26] [SC4MP/Thread-45] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection timed out.
[93m [21:12:45] [SC4MP/Thread-36] (Fetcher) [WARNING] Server timed out.
[93m [21:12:45] [SC4MP/Thread-39] (Fetcher) [WARNING] Server timed out.
[93m [21:12:47] [SC4MP/Thread-45] (Fetcher) [WARNING] Server timed out.
[90m [21:13:50] [SC4MP/Thread-52] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7240...
[90m [21:13:50] [SC4MP/Thread-52] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:50] [SC4MP/Thread-53] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7241...
[90m [21:13:50] [SC4MP/Thread-53] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[91m [21:13:50] [SC4MP/Thread-52] (Fetcher) [ERROR] Connection closed.

Traceback (most recent call last):
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 309, in run
    self.server_list_0_8()
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 571, in server_list_0_8
    servers = recv_json(s)
              ^^^^^^^^^^^^
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\core\networking.py", line 55, in recv_json
    raise ConnectionClosedException()
core.networking.ConnectionClosedException: Connection closed.

[90m [21:13:50] [SC4MP/Thread-54] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7242...
[90m [21:13:50] [SC4MP/Thread-54] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:50] [SC4MP/Thread-55] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7243...
[90m [21:13:50] [SC4MP/Thread-56] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7244...
[90m [21:13:50] [SC4MP/Thread-57] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7245...
[90m [21:13:50] [SC4MP/Thread-57] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:50] [SC4MP/Thread-55] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:50] [SC4MP/Thread-56] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:50] [SC4MP/Thread-58] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7246...
[90m [21:13:51] [SC4MP/Thread-58] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:51] [SC4MP/Thread-59] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7247...
[90m [21:13:51] [SC4MP/Thread-59] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:51] [SC4MP/Thread-60] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7248...
[90m [21:13:51] [SC4MP/Thread-60] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:51] [SC4MP/Thread-61] (Fetcher) [INFO] Fetching server at servers.sc4mp.org:7249...
[91m [21:13:51] [SC4MP/Thread-58] (Fetcher) [ERROR] [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_8kg0yfaa\\Regions\\Downloads\\config.bmp'

Traceback (most recent call last):
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 402, in _calculate_region_stats
    region_dimensions = get_bitmap_dimensions(region_config_path)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 112, in get_bitmap_dimensions
    with open(filename, "rb") as file:
         ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_8kg0yfaa\\Regions\\Downloads\\config.bmp'

[90m [21:13:51] [SC4MP/Thread-62] (Fetcher) [INFO] Fetching server at 136.35.137.78:7240...
[90m [21:13:51] [SC4MP/Thread-63] (Fetcher) [INFO] Fetching server at 35.151.248.28:7240...
[90m [21:13:51] [SC4MP/Thread-62] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:51] [SC4MP/Thread-64] (Fetcher) [INFO] Fetching server at 138.201.94.174:7240...
[90m [21:13:51] [SC4MP/Thread-65] (Fetcher) [INFO] Fetching server at 163.252.250.125:7251...
[90m [21:13:51] [SC4MP/Thread-64] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:52] [SC4MP/Thread-65] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:52] [SC4MP/Thread-66] (Fetcher) [INFO] Fetching server at 24.128.151.227:7240...
[90m [21:13:52] [SC4MP/Thread-67] (Fetcher) [INFO] Fetching server at 163.252.250.125:7240...
[90m [21:13:52] [SC4MP/Thread-68] (Fetcher) [INFO] Fetching server at 163.252.250.125:7246...
[90m [21:13:53] [SC4MP/Thread-68] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.[90m [21:13:53] [SC4MP/Thread-67] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.

[90m [21:13:53] [SC4MP/Thread-69] (Fetcher) [INFO] Fetching server at 163.252.250.125:7245...
[90m [21:13:54] [SC4MP/Thread-70] (Fetcher) [INFO] Fetching server at 163.252.250.125:7247...
[90m [21:13:54] [SC4MP/Thread-70] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:54] [SC4MP/Thread-71] (Fetcher) [INFO] Fetching server at 163.252.250.125:7243...
[90m [21:13:54] [SC4MP/Thread-71] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:54] [SC4MP/Thread-72] (Fetcher) [INFO] Fetching server at 163.252.250.125:7248...
[90m [21:13:54] [SC4MP/Thread-72] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:55] [SC4MP/Thread-73] (Fetcher) [INFO] Fetching server at 163.252.250.125:7242...
[90m [21:13:55] [SC4MP/Thread-73] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:56] [SC4MP/Thread-74] (Fetcher) [INFO] Fetching server at 163.252.250.125:7241...
[90m [21:13:56] [SC4MP/Thread-74] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:56] [SC4MP/Thread-69] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:56] [SC4MP/Thread-75] (Fetcher) [INFO] Fetching server at 67.59.70.25:7240...
[91m [21:13:56] [SC4MP/Thread-68] (Fetcher) [ERROR] [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_bvl134gd\\Regions\\Downloads\\config.bmp'

Traceback (most recent call last):
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 402, in _calculate_region_stats
    region_dimensions = get_bitmap_dimensions(region_config_path)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Users\Kegan\Desktop\Projects\mod\sc4\SC4MP\sc4mp-api\sc4mpapi.py", line 112, in get_bitmap_dimensions
    with open(filename, "rb") as file:
         ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Kegan\\AppData\\Local\\Temp\\sc4mp_api_bvl134gd\\Regions\\Downloads\\config.bmp'

[90m [21:13:58] [SC4MP/Thread-76] (Fetcher) [INFO] Fetching server at 163.252.250.125:7244...
[90m [21:13:58] [SC4MP/Thread-76] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection closed.
[90m [21:13:59] [SC4MP/Thread-75] (Fetcher) [INFO] Connection failed, falling back to legacy protocol: Connection refused by remote host.
```

Do NOT touch the logger itself

--- a/sc4mpapi.py
+++ b/sc4mpapi.py
@@ -304,11 +304,11 @@ class Scanner(Thread):
 
 				except TimeoutError:
 
-					print("[WARNING] Server timed out.")
+					print(f"[WARNING] Server at {self.server[0]}:{self.server[1]} timed out.")
 
 				except Exception as e:
 
-					print(f"[WARNING] Failed to fetch server: {e}")
+					print(f"[WARNING] Failed to fetch server at {self.server[0]}:{self.server[1]}: {e}")
 
 				self.parent.thread_count -= 1
 

--- a/sc4mpapi.py
+++ b/sc4mpapi.py
@@ -23,12 +23,6 @@ try:
 except ImportError:
 	sc4mp_has_flask = False
 
-try:
-	import socks
-	sc4mp_has_socks = True
-except ImportError:
-	sc4mp_has_socks = False
-
 from core.networking import \
 	ClientSocket, NetworkException, ConnectionClosedException, \
 	send_json, recv_json, BUFFER_SIZE
@@ -58,14 +52,9 @@ def init():
 
 def main():
 
-	global sc4mp_args #, sc4mp_proxy
+	global sc4mp_args
 
 	sc4mp_args = parse_args()
-
-	# if None in [sc4mp_args.proxy_host, sc4mp_args.proxy_port]:
-	# 	sc4mp_proxy = None
-	# else:
-	# 	sc4mp_proxy = (sc4mp_args.proxy_host, int(sc4mp_args.proxy_port))
 
 	print("Starting webserver...")
 
@@ -101,8 +90,6 @@ def parse_args():
 
 	parser.add_argument("--host", required=False)
 	parser.add_argument("--port", required=False)
-	parser.add_argument("--proxy-host",  required=False)
-	parser.add_argument("--proxy-port", required=False)
 
 	return parser.parse_args()
 
@@ -336,21 +323,11 @@ class Scanner(Thread):
 			return ClientSocket(address=self.server, timeout=timeout)
 
 
-		def socket_0_8(self, use_proxy=True):
+		def socket_0_8(self):
 			"""Create a regular socket for v0.8/v0.4 protocol"""
-			# if not use_proxy or sc4mp_proxy is None:
 			s = socket()
-			# else:
-			# 	try:
-			# 		s = socks.socksocket()
-			# 		s.set_proxy(socks.SOCKS5, sc4mp_proxy[0], sc4mp_proxy[1])
-			# 	except (socks.SOCKS5Error, socks.GeneralProxyError):
-			# 		return self.socket(use_proxy=False)
-
 			s.settimeout(30)
-
 			s.connect(self.server)
-
 			return s
 
 

--- a/sc4mpapi.py
+++ b/sc4mpapi.py
@@ -277,8 +277,7 @@ class Scanner(Thread):
 					use_legacy = False
 					try:
 						server_id, server_version = self.fetch()
-					except (NetworkException, ConnectionClosedException) as e:
-						print(f"[INFO] Connection failed, falling back to legacy protocol: {e}")
+					except (NetworkException, ConnectionClosedException):
 						use_legacy = True
 
 					# Use legacy protocol if needed
@@ -309,7 +308,7 @@ class Scanner(Thread):
 
 				except Exception as e:
 
-					show_error(e)
+					print(f"[WARNING] Failed to fetch server: {e}")
 
 				self.parent.thread_count -= 1
 

